appinventor/blocklyeditor/src/blocks/procedures.js

```diff
@@ -902,3 +902,49 @@ Blockly.Blocks['procedures_callreturn'] = {
   typeblock: [{ translatedName: Blockly.Msg.LANG_PROCEDURES_CALLRETURN_TRANSLATED_NAME}]
 };
 
+Blockly.Blocks['map_list2'] = {
+  // Map a list with a specific procedure
+  category: 'Procedures',
+  helpUrl: "helpUrl",
+  init: function () {
+    this.setColour(Blockly.PROCEDURE_CATEGORY_HUE);
+    var procDb = this.getTopWorkspace().getProcedureDatabase();
+    var procNameArray = [Blockly.FieldProcedure.defaultValue];
```

Comment:
`var procNameArray = [Blockly.FieldProcedure.defaultValue];` is outside from the closure and inside the function `this.procNamesFxn` you are not clearing the options that are already inside the `procNameArray`.

My suggestion is that you can either clear the array at line 994 as `procNameArray = []` or `procNameArray = [Blockly.FieldProcedure.defaultValue];`.

Reply:
Addressing the closure issue indeed solved the problem, thanks!